`azurerm_dev_center_dev_box_definition` - add HibernateSupport (with `hibernate_support_enabled` property) (#29995)
hashicorp/terraform-provider-azurerm: website/docs/d/dev_center_dev_box_definition.html.markdown

* `sku_name` - The name of the SKU for the Dev Center Dev Box Definition.

> * `hibernate_support_enabled` - Should the Dev Boxes created with this definition be capable of hibernation? Defaults to `false`. Not all images are capable of supporting hibernation. To find out more see https://aka.ms/devbox/hibernate.

Reviewer: Couple minor notes on conventions:
- For data sources, we don't generally document default values
- Additional information beyond a basic description of the property is generally added to a note below the property
- While we have many instances of phrasing the description in the form of a question, these should be considered a legacy format. We'll want to format this as a statement

e.g.

```suggestion
* `hibernate_support_enabled` - Whether Dev Boxes created with this definition are capable of hibernation.

-> **Note:** Not all images are capable of supporting hibernation, for more information see https://aka.ms/devbox/hibernate.
```

Author: Done
